.github/workflows/pullrequests.yml

```diff
@@ -25,8 +26,45 @@ jobs:
             */*/node_modules
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
 
+      - uses: actions/cache@v2
+        id: eslint-cache
+        with:
+          path: |
+            .eslintcache
+            tsconfig.tsbuildinfo
+          key: ${{ runner.os }}-eslint-tsbuildinfo-${{ hashFiles('**/*.ts','**/*.js', 'package.json', 'tsconfig.json') }}
+
       - name: Install NPM Dependencies
         run: npm install
 
       - name: Check for errors
         run: npm run check
+
+  test:
+    name: "Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+          cache: "npm"
+
+      - uses: actions/cache@v2
+        id: node-modules-cache
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Install NPM Dependencies
+        run: npm install
```

Comment:
This should probably be deduped across the 2 jobs, since it's running on the same machine. https://github.com/cloudflare/wrangler2/runs/4842985784?check_suite_focus=true

Reply:
I'll do this in a follow up